algebird-core/src/main/scala/com/twitter/algebird/MomentsGroup.scala

```diff
@@ -108,35 +110,108 @@ object MomentsGroup extends Group[Moments] with CommutativeGroup[Moments] {
 
   override val zero: Moments = Moments(0L, 0.0, 0.0, 0.0, 0.0)
 
-  override def negate(a: Moments): Moments =
-    Moments(-a.count, a.m1, -a.m2, -a.m3, -a.m4)
-
   // Combines the moment calculations from two streams.
   // See http://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Higher-order_statistics
   // for more information on the formulas used to update the moments.
   override def plus(a: Moments, b: Moments): Moments = {
-    val delta = b.mean - a.mean
     val countCombined = a.count + b.count
-    if (countCombined == 0)
-      return zero
```

Comment:
remove the use of return and move `delta` below the else in the case we don't need it.